core/src/main/java/greencity/webcontroller/ManagementFactOfTheDayController.java

```diff
@@ -39,4 +54,114 @@ public String getAllFacts(Model model, @ApiIgnore Pageable pageable) {
         model.addAttribute("languages", languageService.getAllLanguages());
         return "core/management_fact_of_the_day";
     }
+
+    /**
+     * Returns management page with facts of the day that satisfy query.
+     *
+     * @param query string to search
+     * @return model
+     */
+    @ApiPageable
+    @ApiOperation(value = "Get management page with facts of the day that satisfy query.")
+    @GetMapping("/findAll")
+    public String findAll(@RequestParam(required = false, name = "query") String query,
+                          Model model, @ApiIgnore Pageable pageable) {
+        PageableDto<FactOfTheDayDTO> pageableDto = query == null || query.isEmpty()
+            ? factOfTheDayService.getAllFactsOfTheDay(pageable) : factOfTheDayService.searchBy(pageable, query);
+        model.addAttribute("pageable", pageableDto);
+        model.addAttribute("languages", languageService.getAllLanguages());
+        return "core/management_fact_of_the_day";
+    }
+
+    /**
+     * Method which saves {@link FactOfTheDay}.
+     *
+     * @param factOfTheDayPostDTO of {@link FactOfTheDayPostDTO}
+     * @return {@link GenericResponseDto} with of operation and errors fields
+     */
+    @ApiOperation(value = "Save fact of the day")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK, response = GenericResponseDto.class),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @ResponseBody
+    @PostMapping("/")
+    public GenericResponseDto saveFactOfTheDay(@Valid @RequestBody FactOfTheDayPostDTO factOfTheDayPostDTO,
+                                               BindingResult bindingResult) {
+        if (bindingResult.hasErrors()) {
+            GenericResponseDto genericResponseDto = new GenericResponseDto();
+            for (FieldError fieldError : bindingResult.getFieldErrors()) {
+                genericResponseDto.getErrors().add(
+                    new FieldErrorDto(fieldError.getField(), fieldError.getDefaultMessage()));
+            }
+            return genericResponseDto;
+        }
+        factOfTheDayService.saveFactOfTheDayAndTranslations(factOfTheDayPostDTO);
+        return GenericResponseDto.builder().errors(new ArrayList<>()).build();
```

Comment:
I think you don't need `.errors(new ArrayList<>())`

Reply:
Okay